### PR TITLE
CNFT2-1753 filter duplicate / Inactive questions

### PR DESCRIPTION
**Pull request by mpeels**
## Description

When searching for questions to add to a page, the API should not return questions that are already present on the page or have a status of `Inactive`.

This PR does a lot. I'm sorry

1. Creates a new API to drive the question search for a page
2. Refactors the `add question to page` API to allow adding multiple questions at once
3. Updates the `AddQuestionModal` to match the latest Figma designs
4. Adds a `refresh` callback to the `usePageManagement` hook to allow "soft" refreshing of the page without triggering a full load.

## Tickets

* [CNFT2-1753](https://cdc-nbs.atlassian.net/browse/CNFT2-1753)
### Showing table change and adding a question
addToPageModal

### Showing a question is hidden if Inactive

hideInactive

## Checklist before requesting a review
- [x] PR focuses on a single story
- [x] Code has been fully tested to meet acceptance criteria
- [ ] PR is reasonably small and reviewable (Generally less than 10 files and 500 changed lines)
- [x] All new functions/classes/components reasonably small
- [x] Functions/classes/components focused on one responsibility
- [x] Code easy to understand and modify (clarity over concise/clever)
- [x] PRs containing TypeScript follow the [Do's and Don'ts](https://www.typescriptlang.org/docs/handbook/declaration-files/do-s-and-don-ts.html)
- [x] PR does not contain hardcoded values (Uses constants)
- [ ] All code is covered by unit or feature tests

[CNFT2-1753]: https://cdc-nbs.atlassian.net/browse/CNFT2-1753?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: apps/modernization-ui/src/apps/page-builder/page/management/edit/add-question/search/question-search.module.scss [hunk 1]
```diff
@@ -0,0 +1,51 @@
+@use 'styles/borders';
+@use 'styles/colors';
+
+.content {
+    padding: 0 1.5rem;
+    display: flex;
+    min-height: 100px;
```
Comment (lramakr) [inline apps/modernization-ui/src/apps/page-builder/page/management/edit/add-question/search/question-search.module.scss:7]: px to rem

File: apps/modernization-ui/src/apps/page-builder/components/TabbedButtonBar/tabbed-button-bar.module.scss [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+        &.hover {
+            border: none;
+        }
+    }
+    .active {
+        color: black !important;
+        font-weight: 700;
+        border-radius: 0.25rem;
+        background: colors.$base-white;
+        box-shadow: 0px 2px 2px 0px rgba(0, 0, 0, 0.1);
```
Comment (lramakr) [inline apps/modernization-ui/src/apps/page-builder/components/TabbedButtonBar/tabbed-button-bar.module.scss:29]: px to rem

File: apps/modernization-ui/src/apps/page-builder/page/management/edit/add-question/search/table/expanded-question.module.scss [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+    .content {
+        display: flex;
+        padding: 1.5rem;
+        flex-wrap: wrap;
+        gap: 1.5rem;
+        align-items: flex-start;
+        flex-shrink: 0;
+        border-radius: 0.3125rem;
+        border: 1px solid colors.$base-lighter;
```
Comment (lramakr) [inline apps/modernization-ui/src/apps/page-builder/page/management/edit/add-question/search/table/expanded-question.module.scss:18]: px to rem

File: apps/modernization-ui/src/apps/page-builder/page/management/edit/add-question/search/question-search.module.scss (hunk 1 above)
Comment (mpeels) [reply apps/modernization-ui/src/apps/page-builder/page/management/edit/add-question/search/question-search.module.scss:7]: good catch. looks like i can remove this min-height completely

File: apps/modernization-ui/src/apps/page-builder/components/TabbedButtonBar/tabbed-button-bar.module.scss (hunk 2 above)
Comment (mpeels) [reply apps/modernization-ui/src/apps/page-builder/components/TabbedButtonBar/tabbed-button-bar.module.scss:29]: for the box shadow, we don't want the shadow size to change when the user zooms in/out, so it needs to stay px.

File: apps/modernization-ui/src/apps/page-builder/page/management/edit/add-question/search/table/expanded-question.module.scss (hunk 3 above)
Comment (mpeels) [reply apps/modernization-ui/src/apps/page-builder/page/management/edit/add-question/search/table/expanded-question.module.scss:18]: removing this border-radius and border styles and leveraging the pre-defined `styles/borders`

**Review by lramakr — APPROVED** [review]
LGTM

**Review by alaapbharadwaj — APPROVED** [review]

**Pull request merged**